jumanji/environments/packing/tetris/viewer.py

```diff
@@ -0,0 +1,328 @@
+        if render_mode == "rgb_array":
+            self._display = self._display_rgb_array
+        elif render_mode == "human":
+            self._display = self._display_human
+        else:
+            raise ValueError(f"Invalid render mode: {render_mode}")
+
+        # Pick colors.
+        colormap_indecies = jnp.arange(0, 1, 1 / self.n_colors)
+        colormap = matplotlib.cm.get_cmap("hsv", self.n_colors + 1)
+
+        self.colors = [(1.0, 1.0, 1.0, 1.0)]  # Initial color must be white.
+        for colormap_idx in colormap_indecies:
+            self.colors.append(colormap(colormap_idx))
+        self.edgecolors = [(0.0, 0.0, 0.0), (0.9, 0.9, 0.9)]
+
+        # The animation must be stored in a variable that lives as long as the
+        # animation should run. Otherwise, the animation will get garbage-collected.
+        self._animation: Optional[matplotlib.animation.Animation] = None
+
+    def render(self, state: State) -> Optional[NDArray]:
+        """Render Tetris.
+
+        Args:
+            grid: the grid of the Tetris environment to render.
+
+        Returns:
+            RGB array if the render_mode is RenderMode.RGB_ARRAY.
+        """
+        self._clear_display()
+        fig, ax = self._get_fig_ax()
+        ax.clear()
+        fig.suptitle(f"Tetris    Score: {int(state.score)}", size=20)
+        ax.invert_yaxis()
+        grid = self._create_rendering_grid(state)
+        self._add_grid_image(ax, grid)
+        return self._display(fig)
+
+    def _move_tetromino(
+        self, state: State, old_padded_grid: chex.Array
+    ) -> List[chex.Array]:
+        """shifts the tetromino from center to the selected position
+
+        Args:
+            state: `State` object containing the current environment state.
+            old_padded_grid: `chex.Array` containing the grid before placing the tetromino.
+
+        Returns:
+            grids: `List[NDArray]` contais a list of grids.
+        """
+        grids = []
+        grid = old_padded_grid[: self.num_rows, : self.num_cols]
+        center_position = self.num_cols - 4
+        # step is 1 to move to the right and -1 to move to the left
+        step = 1 if center_position < state.x_position else -1
+        for xi in range(center_position, state.x_position + step, step):
+            tetromino_zonne = jnp.zeros((4, state.grid_padded.shape[1]))
+            tetromino_zonne = tetromino_zonne.at[0:4, xi : xi + 4].add(
+                state.old_tetromino_rotated
+            )
+            # delete the cols dedicated for the right padding
+            tetromino_zonne = tetromino_zonne[:, : self.num_cols]
+            # Stack the tetromino with grid position
+            mixed_grid = jnp.vstack((tetromino_zonne, grid))
+            grids.append(mixed_grid)
+        return grids
+
+    def _crush_lines(
+        self, state: State, grid: chex.Array, n: int = 2
+    ) -> List[chex.Array]:
+        """Creates annimation when a line is crushed by togling its value.
```

Comment:
```suggestion
        """Creates animation when a line is crushed by toggling its value.
```
If you are using PyCharm, I think you can enable a dictionary checker to help if there are other places.